### PR TITLE
add constraint to ensure woopool quote is in quote token

### PR DESCRIPTION
**Pull request by toprince**
PR for audit issue 40 & 64

https://github.com/sherlock-audit/2024-08-woofi-solana-deployment-judging/issues/40
https://github.com/sherlock-audit/2024-08-woofi-solana-deployment-judging/issues/64

add constraint to ensure woopool quote is in quote token

### CODE REVIEW

File: programs/woofi/src/instructions/query.rs [hunk 1]
```diff
@@ -57,6 +57,8 @@ pub struct Query<'info> {
     price_update_to: Account<'info, PriceUpdateV2>,
 
     #[account(
+        has_one = wooconfig,
+        constraint = woopool_quote.token_mint == woopool_quote.quote_token_mint,
         constraint = woopool_quote.token_mint == woopool_from.quote_token_mint,
```
Comment (fb-alexcq) [inline programs/woofi/src/instructions/query.rs:62]: Also need add check:

  woopool_quote.token_mint == woopool_to.quote_token_mint
  
  ?

File: programs/woofi/src/instructions/swap.rs [hunk 2]
```diff
@@ -78,6 +78,7 @@ pub struct Swap<'info> {
 
     #[account(mut,
         has_one = wooconfig,
+        constraint = woopool_quote.token_mint == woopool_quote.quote_token_mint,
```
Comment (fb-alexcq) [inline programs/woofi/src/instructions/swap.rs:81]: ditto

File: programs/woofi/src/instructions/query.rs (hunk 1 above)
Comment (toprince) [reply programs/woofi/src/instructions/query.rs:62]: though it has check above, but added here should be more clear.
Added.

File: programs/woofi/src/instructions/swap.rs (hunk 2 above)
Comment (toprince) [reply programs/woofi/src/instructions/swap.rs:81]: same

- [ ]

**Pull request merged**